### ISSUE TITLE
Use the links argument in the family constructed by the models

### ISSUE THREAD

**Issue by venpopov** [venpopov/bmm#189]
We have the links argument setup for all models, but we currently do not pass that information to the family. We should add that functionality to the existing models.

**Comment by venpopov** [comment]
decision - links just for brms parameters we can pass to a family for now.

**Comment by venpopov** [comment]
now that we have the apply_links function from #237 this should be straightforward to do

**Comment by GidonFrischkorn** [comment]
I agree that this should now be straightforward for both models specified by complex `brmsformulae` and `custom_families`. The question is if allowing users to use their own links is reasonable for all models. Specifically:

- in the `imm` all parameters use `log` links to ensure that they are positive with is necessary given the non-exponentiated version of Luce's choice rule we use there
- in the `mixture2p` and `mixture3p` model `kappa` uses a log link which I think is reasonable given it should be positive and the mixture weights `thetat` and `thetant` are internally converted to probabilities using the `softmax` function so there is no real reason to apply links to these parameters
- in the `sdm` it is technically easy to set links if you wrap the call to the `custom_family` function in a function setting the links for the family (see example below). However the `log` link of `c` is hard coded into the PDF if I recall correctly

So from my perspective the links are really mostly relevant in the `m3` where specific parameter might have different ranges on the native scale and thus would require different links that users can set. If we want we could allow users to set links that are less optimal with respect to parameter estimation. 

I see how there might be some cases where users want to specify more complex non-linear formulas and choose the `links` themselves. But we could also wait and see how often this actually happens and if someone raises this issue.

```
# set up the custom family
   ezdm_family <- function(link = "identity", link_drift = "identity",
                    link_bound = "log", link_ndt = "log") {
      brms::custom_family(
         "ezdm",
         dpars = c("mu", "drift", "bound", "ndt"), # Those will be estimated
         links = c(link, link_drift, link_bound, link_ndt),
         lb = c(NA, NA, 0, 0), # bounds for the parameters
         ub = c(NA, NA, NA, NA),
         vars = c("vreal1[n]" , "vint1[n]", "trials[n]"),
         loop = TRUE,
         type = "real"
      )
   }
```

**Comment by GidonFrischkorn** [comment]
One option that just came to my mind for the `mixture2p` and `mixture3p` models is that we could see if it is possible to constraining the sum of `thetat + thetant + thetag = 1` together with implementing a `simple` choice rule like we did for the `m3` this would allow to estimate effects on the probability scale. In this case using a `logit` link for `theta` and `thetant` might be reasonable. But I think this would mainly make a difference for the `mixture3p` model as the interpretation of parameters for the `mixture2p` model is pretty straightforward with the current implementation.

**Comment by GidonFrischkorn** [comment]
As discussed in #328, we need to find a consistent solution for the use of the `links` specified in the `bmmodel` object across all `bmmodels`. The current status is:

- `mixture2p`, `mixture3p`, `imm`, and `sdm` do not use the links when specifying the model and pass no information to `brms` so that the link information is primarily an information for users
- `m3`, `cswald`, `ezdm`, and `ddm` use the information in `links` during model fitting and thus changing them also affects the parameter estimation.

We need to find a more coherent solution to ease maintainability and avoid that this inconsistency creates problems in the future.

**Comment by GidonFrischkorn** [comment]
This is scoped as "pass the links to the family," but the real problem is wider: the links live in three places that disagree, so harmonizing them means one source of truth rather than one more copy.

- `link_transform` (init values, conditional-effects back-transform) supports 11 links ([helpers-parameters.R:86-118](R/helpers-parameters.R#L86-L118)).
- `inv_link` (injects the inverse-link expression for non-linear M3 formulas) supports only 4, via `match.arg(c("log", "logit", "probit", "identity"))` ([bmmformula.R:451-461](R/bmmformula.R#L451-L461)).
- `check_model.m3_custom` defines a third, implicit allow-list of 3 links inside its prior switch ([model_m3.R:234-251](R/model_m3.R#L234-L251)).

Two failures fall out of this. A link that is valid for init (e.g. `sqrt`) errors the moment that parameter appears in a non-linear formula, because `inv_link` rejects it. And the custom-family models build their `links` argument as a positional vector hand-aligned to `dpars` — e.g. `links = c("identity", link_drift, link_bound, link_ndt, link_zr)` ([model_ddm.R:253](R/model_ddm.R#L253)) — so a reordering is a silent bug.

The broader fix is a single link registry: each entry carries the forward transform, the numeric inverse (init and conditional-effects), and the Stan inverse-link expression (`apply_links`). `link_transform`, `inv_link`, the M3 prior switch, and the custom-family `links` argument all read from it, and `supported_links()` becomes a real function. That also removes the per-link tax that #363 (softplus) is currently paying. The existing links keep identical math, so no fit changes — guard with a snapshot of `inv_link` output for the current four.
